Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Services/NMDataServiceTest.cs

```diff
@@ -0,0 +1,196 @@
+﻿using FakeItEasy;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using UKHO.FileShareClient.Models;
+using UKHO.MaritimeSafetyInformation.Common.Configuration;
+using UKHO.MaritimeSafetyInformation.Common.Helpers;
+using UKHO.MaritimeSafetyInformation.Common.Models.NoticesToMariners;
+using UKHO.MaritimeSafetyInformation.Web.Services;
+using UKHO.MaritimeSafetyInformation.Web.Services.Interfaces;
+
+namespace UKHO.MaritimeSafetyInformation.Web.UnitTests.Services
+{
+    [TestFixture]
+    public class NMDataServiceTest
+    {
+        private IFileShareService _fakefileShareService;
+        private ILogger<NMDataService> _fakeLogger;
+        private IAuthFssTokenProvider _fakeAuthFssTokenProvider;
+        private NMDataService _fakeNMDataService;
+        public const string CorrelationId = "7b838400-7d73-4a64-982b-f426bddc1296";
+
+        [SetUp]
+        public void Setup()
+        {
+            _fakefileShareService = A.Fake<IFileShareService>();
+            _fakeLogger = A.Fake<ILogger<NMDataService>>();
+            _fakeAuthFssTokenProvider = A.Fake<IAuthFssTokenProvider>();
+
+            _fakeNMDataService = new NMDataService(_fakefileShareService, _fakeLogger, _fakeAuthFssTokenProvider);
+        }
+
+
+        [Test]
+        public void WhenGetNMBatchFilesIsCalled_ThenShouldReturnsMoreThanZeroFiles()
+        {
+            int year = 2022;
+            int week = 15;
+
+            A.CallTo(() => _fakeAuthFssTokenProvider.GenerateADAccessToken(A<string>.Ignored));
+
+            Result<BatchSearchResponse> SearchResult = SetSearchResult();
+
+            A.CallTo(() => _fakefileShareService.FssBatchSearchAsync(A<string>.Ignored, A<string>.Ignored, CorrelationId)).Returns(SearchResult);
+
+            string expectedstatus = "RanToCompletion";
+
+            Task<List<ShowFilesResponseModel>> ListshowFilesResponseModels = _fakeNMDataService.GetWeeklyBatchFiles(year, week, CorrelationId);
+
+            Assert.AreEqual(expectedstatus, ListshowFilesResponseModels.Status.ToString());
+        }
```

Comment:
What is this test actually testing?

The asserts checks the status of the Task but doesn't inspect the content of the Response!

Im assuming in this case we should check that the Response contains 2 entries?

Reply:
Assert changed to check if the response contains the expected no of entries.